src/ORM/DataObject.php

```diff
@@ -203,6 +203,12 @@ class DataObject extends ViewableData implements DataObjectInterface, i18nEntity
      */
     const CREATE_HYDRATED = 2;
 
+    /**
+     * Value for 2nd argument to constructor, indicating that a record is being hydrated from memory. This can be used
+     * to initialised a record that doesn't yet have an ID. Neither setters and nor default population will be called.
```

Comment:
Sentence could be structured a bit more clearly - e.g. `Setter methods are not called, and population via private static $defaults will not occur`.